pkg/reconciler/v1alpha1/broker/broker.go

```diff
@@ -0,0 +1,442 @@
+var _ reconcile.Reconciler = &reconciler{}
+
+type ReconcilerArgs struct {
+	IngressImage              string
+	IngressServiceAccountName string
+	FilterImage               string
+	FilterServiceAccountName  string
+}
+
+// ProvideController returns a function that returns a Broker controller.
+func ProvideController(logger *zap.Logger, args ReconcilerArgs) func(manager.Manager) (controller.Controller, error) {
+	return func(mgr manager.Manager) (controller.Controller, error) {
+		// Setup a new controller to Reconcile Brokers.
+		c, err := controller.New(controllerAgentName, mgr, controller.Options{
+			Reconciler: &reconciler{
+				recorder: mgr.GetRecorder(controllerAgentName),
+				logger:   logger,
+
+				ingressImage:              args.IngressImage,
+				ingressServiceAccountName: args.IngressServiceAccountName,
+				filterImage:               args.FilterImage,
+				filterServiceAccountName:  args.FilterServiceAccountName,
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Watch Brokers.
+		if err = c.Watch(&source.Kind{Type: &v1alpha1.Broker{}}, &handler.EnqueueRequestForObject{}); err != nil {
+			return nil, err
+		}
+
+		// Watch all the resources that the Broker reconciles.
+		for _, t := range []runtime.Object{&v1alpha1.Channel{}, &corev1.Service{}, &v1.Deployment{}} {
+			err = c.Watch(&source.Kind{Type: t}, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.Broker{}, IsController: true})
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		return c, nil
+	}
+}
+
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	return nil
+}
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Broker resource
+// with the current status of the resource.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
+	ctx = logging.WithLogger(ctx, r.logger.With(zap.Any("request", request)))
+
+	broker := &v1alpha1.Broker{}
+	err := r.client.Get(ctx, request.NamespacedName, broker)
+
+	if errors.IsNotFound(err) {
+		logging.FromContext(ctx).Info("Could not find Broker")
+		return reconcile.Result{}, nil
+	}
+
+	if err != nil {
+		logging.FromContext(ctx).Error("Could not Get Broker", zap.Error(err))
+		return reconcile.Result{}, err
+	}
+
+	// Reconcile this copy of the Broker and then write back any status updates regardless of
+	// whether the reconcile error out.
+	result, reconcileErr := r.reconcile(ctx, broker)
+	if reconcileErr != nil {
+		logging.FromContext(ctx).Error("Error reconciling Broker", zap.Error(reconcileErr))
+	} else if result.Requeue || result.RequeueAfter > 0 {
+		logging.FromContext(ctx).Debug("Broker reconcile requeuing")
+	} else {
+		logging.FromContext(ctx).Debug("Broker reconciled")
+		r.recorder.Event(broker, corev1.EventTypeNormal, brokerReconciled, "Broker reconciled")
+	}
+
+	if _, err = r.updateStatus(broker.DeepCopy()); err != nil {
```

Comment:
No need to DeepCopy. controller-fwk's Get() returns a deep copied object

Reply:
Done.